en/docs/observability-and-monitoring/supported-observability-tools-and-platforms/opensearch.md

```diff
@@ -0,0 +1,726 @@
+
+```toml
+[build-options]
+cloud = "docker"
+```
+
+* Create a file named `Cloud.toml` in the project root directory and add the following content.
+
+```toml
+[container.image]
+repository="wso2inc"
+name="sample-service"
+tag="v0.1.0"
+
+[settings]
+buildImage=true
+```
+
+* In the `BI view`, under the `Deployment Options` select `Deploy on a VM` and select `Create Executable` option. This will create a Docker image `wso2inc/sample-service:v0.1.0`.
+
+## Step 2 - Setup OpenSearch Deployment
+
+Main components:
+    * Sample Service: service application that is being observed.
+    * OpenSearch: Stores and indexes logs/traces; exposes API and metrics ports.
+    * OpenSearch Dashboards: Visualizes data from OpenSearch via web UI.
+    * Data Prepper: Receives and processes OpenTelemetry data, sends it to OpenSearch.
+    * Fluent Bit: Collects app logs and forwards them for indexing.
+    * Setup Container: Automates initial setup—creates index templates and imports dashboards.
+
+**Sample Service**
+
+Create a `Config.toml` file in the `config/ballerina` directory to enable metrics, logs, and traces in the application and to configure the Jaeger exporter. Ensure the directory structure matches `config/ballerina/Config.toml`.
+
+```
+[ballerina.observe]
+metricsLogsEnabled = true
+tracingEnabled = true
+tracingProvider = "jaeger"
+
+[ballerinax.jaeger]
+agentHostname = "data-prepper"
+agentPort = 4317
+samplerType = "const"
+samplerParam = 1.0
+reporterFlushInterval = 2000
+reporterBufferSize = 1000
+
+[ballerinax.metrics.logs]
+logFilePath = "/var/log/ballerina/sample-service/app.log"
+```
+
+**Data Prepper**
+
+Create `config/data-prepper/pipelines.yaml` to store the data prepper configurations.
+
+```
+entry-pipeline:
+  delay: "100"
+  source:
+    otel_trace_source:
+      port: 4317
+      ssl: false
+  sink:
+    - pipeline:
+        name: "raw-pipeline"
+    - pipeline:
+        name: "service-map-pipeline"
+    - stdout: {}
+
+raw-pipeline:
+  source:
+    pipeline:
+      name: "entry-pipeline"
+  processor:
+    - otel_traces:
+  sink:
+    - opensearch:
+        hosts: [ "https://opensearch:9200" ]
+        username: "admin"
+        password: "Ballerina@123"
+        insecure: true        
+        index_type: trace-analytics-raw
+
+service-map-pipeline:
+  delay: "100"
+  source:
+    pipeline:
+      name: "entry-pipeline"
+  processor:
+    - service_map:
+  sink:
+    - opensearch:
+        hosts: [ "https://opensearch:9200" ]
+        username: "admin"
+        password: "Ballerina@123"
+        insecure: true
+        index_type: trace-analytics-service-map
+```
+
+**Fluent Bit**
+
+Create `config/fluent-bit/fluent-bit.conf` and `config/fluent-bit/parsers.conf` to store the fluent bit configurations.
+
+* `config/fluent-bit/fluent-bit.conf`
+
+Collects, enriches, and routes Ballerina logs and metrics to OpenSearch for analysis.
```

Comment:
Better indent this sentence and the code snippet below to match with the bullet point, check all the places.